component/authenticator/src/test/java/org/wso2/carbon/identity/authenticator/smsotp/test/SMSOTPAuthenticatorTest.java

```diff
@@ -581,7 +581,6 @@ public void testCheckWithBackUpCodes() throws Exception {
                 context,"1234",user);
     }
 
-    @Test(expectedExceptions = {AuthenticationFailedException.class})
```

Comment:
This exception will be thrown in this test case right? why have we removed this line then?

Reply:
There is a `code mismatch`. Earlier implementation has thrown an AuthenticationFailedException, but now we only set `context.setProperty(SMSOTPConstants.CODE_MISMATCH, true);`. This method doesn't throw an exception for code mismatch. That error throwing is delegated to `processAuthenticationResponse` method.

Reply:
Got it (Y)